### PR TITLE
React 16/Enzyme 3 - Fix datetime tests

### PR DESCRIPTION
**Pull request by cmslewis**
Fix all the datetime tests that broke in the React 16 / Enzyme 3 upgrade.

### CODE REVIEW

File: packages/datetime/test/dateRangeInputTests.tsx [hunk 1]
```diff
@@ -408,24 +412,25 @@ describe("<DateRangeInput>", () => {
...
-            const startInput = getStartInput(root);
-            startInput.simulate("focus");
-            startInput.simulate("change", { target: { value: START_STR } });
-            startInput.simulate("keydown", { which: Keys.ENTER });
+            // Don't save the input elements into variables; they can become
+            // stale across React updates.
+
+            getStartInput(root).simulate("focus");
+            getStartInput(root).simulate("change", { target: { value: START_STR } });
+            getStartInput(root).simulate("keydown", { which: Keys.ENTER });
```
Comment (giladgray) [inline packages/datetime/test/dateRangeInputTests.tsx]: you might be able to chain these simulates and only get a new reference on line 426?

Comment (cmslewis) [reply packages/datetime/test/dateRangeInputTests.tsx]: Chatted IRL. Will leave it b/c we're not sure exactly if the input node would get stale between events.

**Comment by cmslewis** [comment]
Builds failing:

**test-select:**
Failing packages:ReferenceError: Can't find variable: Map

**test-core:**
Coverage for statements (28%) does not meet per-file (/home/circleci/project/packages/core/src/common/abstractComponent.ts)  threshold (80%)
Coverage for lines (29.17%) does not meet per-file (/home/circleci/project/packages/core/src/common/abstractComponent.ts)  threshold (80%)

**Comment by cmslewis** [comment]
Merging b/c the `datetime` tests are fixed.

**Pull request merged**